### PR TITLE
Fix: GLSL Integer Constant Support: constant attribute setter (needs tests)

### PR DESCRIPTION
**Pull request by ibgreen**
For #

#### Change List
-

### CODE REVIEW

File: modules/core/src/webgl/classes/vertex-array-object.js [hunk 1]
```diff
@@ -239,18 +247,19 @@ export default class VertexArrayObject extends Resource {
 
   static _setConstantIntArray(gl, location, array) {
     assert(isWebGL2(gl));
+    // NOTE - array needs to be converted to Int32Array and padded to length 4
```
Comment (tsherif) [inline modules/core/src/webgl/classes/vertex-array-object.js]: This is interesting. What's the background for this requirement? If there's something in the spec or a bug report that explains it, a link here would be helpful.

File: modules/core/src/webgl/classes/vertex-array-object.js [hunk 2]
```diff
@@ -239,18 +247,19 @@ export default class VertexArrayObject extends Resource {
...
     switch (array.length) {
       case 1:
-        gl.vertexAttribI1iv(location, array);
+        gl.vertexAttribI4iv(location, new Int32Array([array[0], 0, 0, 0]));
```
Comment (tsherif) [inline modules/core/src/webgl/classes/vertex-array-object.js]: Pre-allocate the typed arrays.

File: modules/core/src/webgl/classes/vertex-array-object.js (hunk 1 above)
Comment (tsherif) [reply modules/core/src/webgl/classes/vertex-array-object.js]: Oh so those functions just don't exist... Here's the API doc link: https://www.khronos.org/registry/OpenGL-Refpages/es3.0/html/glVertexAttrib.xhtml

Are we using int attributes anywhere?

**Comment by ibgreen** [comment]
@tsherif I believe that your long-term goal is to remove constant attribute support from luma.gl. In the mean-time, should we land PRs like that this that fixes bugs in the current system?

**Comment by tsherif** [comment]
Yeah, this should definitely be fixed since it's currently a trap for anyone who tries to use it. I'd be ok with just removing constant int attribute support, or updating this PR with pre-allocated int/uint arrays and tests.

**Pull request closed without merging**